server/libs/infra/src/utils/pagination.util.ts

```diff
@@ -0,0 +1,44 @@
+import { FindOneOptions, ObjectLiteral, Repository } from 'typeorm';
+
+export interface PaginationOptions {
+  take: number;
+  skip?: number;
+}
+
+export interface PaginationResult<T> {
+  items: T[];
+  hasNextPage: boolean;
+}
+
+export type Paginated<T> = Promise<PaginationResult<T>>;
+
+export async function paginate<Entity extends ObjectLiteral>(
+  repository: Repository<Entity>,
+  paginationOptions: PaginationOptions,
+  searchOptions?: FindOneOptions<Entity>,
+): Paginated<Entity> {
+  const items = await repository.find({
+    ...searchOptions,
+    // Take one more item to check if there's a next page
+    take: paginationOptions.take + 1,
+    skip: paginationOptions.skip,
+  });
+
+  const hasNextPage = items.length > paginationOptions.take;
+  items.splice(paginationOptions.take);
+
+  return { items, hasNextPage };
+}
+
+export async function* usePagination<T>(
```

Comment:
is `*` equivalent to a stream syntax in TS?

Reply:
The `*` is used to define a generator, in this case an async generator which allows for iteration using the `for await...of` syntax. Here's some documentation about this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function*